GCTC-NTGC/gc-digital-talent
# Feature 2640 - admin user api table search

Resolves #2640 

## Summary

- updates the API-Driven table to allow searching by a specific column. 
- Adds the three expected columns to User Table
- Adds new search form not reliant on `react-table`
- Downgrade @reach libraries to prevent the import warning (0.17.0 requires react@18)

## Testing

1. Navigate to `/admin/users`
2. Use the search field + column dropdown
3. Ensure the column dropdown functions as expected
